chainercv/functions/psroi_pooling_2d.py

```diff
@@ -0,0 +1,331 @@
+# Cuda Kernel Original work by Haozhi Qi (@Oh233)
+# https://github.com/msracver/FCIS
+# Modified by Shingo Kitagawa (@knorth55)
+
+import numpy as np
+import six
+
+from chainer.backends import cuda
+from chainer import function
+from chainer.utils import type_check
+
+if cuda.available:
+    import cupy as cp
+
+
+def _roi_pooling_slice(size, stride, max_size, roi_offset):
+    start = int(np.floor(size * stride))
+    end = int(np.ceil((size + 1) * stride))
+
+    start = min(max(start + roi_offset, 0), max_size)
+    end = min(max(end + roi_offset, 0), max_size)
+
+    return slice(start, end), end - start
+
+
+class PSROIPooling2D(function.Function):
+
+    def __init__(self, outd, outh, outw, spatial_scale, group_size):
+        self.outd, self.outh, self.outw = outd, outh, outw
+        self.spatial_scale = spatial_scale
+        self.group_size = group_size
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 2)
+
+        x_type, roi_type = in_types
+        type_check.expect(
+            x_type.dtype == np.float32,
+            x_type.ndim == 4,
+            roi_type.dtype == np.float32,
+            roi_type.ndim == 2,
+            roi_type.shape[1] == 5,
+        )
+
+    def forward_cpu(self, inputs):
+        self.retain_inputs((1,))
+        self._bottom_data_shape = inputs[0].shape
+
+        bottom_data, bottom_rois = inputs
+        channels, height, width = bottom_data.shape[1:]
+        n_rois = bottom_rois.shape[0]
```

Comment:
Could you follow ChainerCV's naming convention? (`n_{Singular}` instead of `n_{Plural}`)
`n_rois` --> `n_roi`